base/permuteddimsarray.jl

```diff
@@ -253,6 +253,16 @@ end
     P
 end
 
+function Base._mapreduce_dim(f, op, init::Base._InitialValue, A::PermutedDimsArray, dims::Colon)
+    Base._mapreduce_dim(f, op, init, parent(A), dims)
+end
+
+function Base.mapreducedim!(f, op, B::AbstractArray{T,N}, A::PermutedDimsArray{T,N,P,Q}) where {T,N,P,Q}
+    C = PermutedDimsArray{T,N,Q,P,typeof(B)}(B)
```

Comment:
EDIT: I think I understand why it is legal to swap these cases. It seems intuitive anyways, even I can't prove it. Can you change `P` and `Q` to `perm` and `iperm`, as I see them named elsewhere to make this clearer here?
```suggestion
function Base.mapreducedim!(f, op, B::AbstractArray{T,N}, A::PermutedDimsArray{T,N,perm,iperm}) where {T,N,perm,iperm}
    C = PermutedDimsArray{T,N,iperm,perm,typeof(B)}(B) # make the inverse permutation for the output
```

Reply:
Sure. I thought I was following Base, but in fact it seems I was following my package... will change.